catalog/github/webhook.js

```diff
@@ -88,6 +88,66 @@ function main(params) {
     });
 
     return whisk.async();
+  } else if(lifecycleEvent === 'DELETE') {
+    //list all the existing webhooks first.
+    var options = {
+        method: 'GET',
+        url: registrationEndpoint,
+        headers: {
+            'Content-Type': 'application/json',
+            'Authorization': authorizationHeader,
+            'User-Agent', 'whisk'
+        }
+    }
+
+    request(options, function(error, response, body){
+      if (error){
+        whisk.error({
+          response: response,
+          error:error,
+          body:body
+        });
+      } else {
+        for(i=0; i<body.length: i++){
+            var urlpath = encodeURIComponent(body[i].config.url);
+            var pathname = url.parse(urlpath).pathname;
+            //if the url contains the triggerName[1], then delete the webhook
+            //so this delete all the trigger start with triggerName[1] such as
+            // trigger1, trigger2.
+            if (pathname.search(encodeURIComponent(triggerName[1])) != -1) {
```

Comment:
Which brings up an interesting idea... Instead of parsing the webhook's `config.url` property to try to pick out the namespace and trigger name, I wonder if you could simply compare it for equality with the `whiskCallbackUrl` variable?